doc/BUILDING.md

```diff
@@ -181,23 +146,24 @@ behaviour of the build.
 
 | Option                                            | Description                                                           | Default |
 |---------------------------------------------------|-----------------------------------------------------------------------|---------|
-| `BUILD_SHARED_LIBS`                               | `ON` to build shared libraries. `OFF` to build static libraries.      | `ON`    |
 | `OPENASSETIO_ENABLE_PYTHON`                       | Additionally build python bindings                                    | `ON`    |
+| `OPENASSETIO_PYTHON_SITEDIR`                      | Python site-packages install prefix subdirectory                      | auto    |
+| `OPENASSETIO_ENABLE_PYTHON_STUBGEN`               | Enable Python .pyi stub file generation                               | `ON`    |
 | `OPENASSETIO_ENABLE_PYTHON_INSTALL_DIST_INFO`     | Create a dist-info metadata directory alongside Python installation   | `ON`    |
-| `OPENASSETIO_ENABLE_C`                            | Additionally build C bindings                                         | `OFF`   |
+| `OPENASSETIO_ENABLE_SIMPLECPPMANAGER`             | Build the SimpleCppManager example plugin                             | `OFF`   |
 | `OPENASSETIO_ENABLE_TESTS`                        | Additionally build tests                                              | `OFF`   |
 | `OPENASSETIO_ENABLE_PYTHON_TEST_VENV`             | Automatically create environment when running tests                   | `ON`    |
 | `OPENASSETIO_WARNINGS_AS_ERRORS`                  | Treat compiler warnings as errors                                     | `OFF`   |
 | `OPENASSETIO_ENABLE_IPO`                          | Enable Interprocedural Optimization, aka Link Time Optimization (LTO) | `ON`    |
 | `OPENASSETIO_ENABLE_POSITION_INDEPENDENT_CODE`    | Enable position independent code for static library builds            | `ON`    |
 | `OPENASSETIO_GLIBCXX_USE_CXX11_ABI`               | When building under gcc, use the new C++11 library ABI                | `OFF`   |
 | `OPENASSETIO_ENABLE_NEW_DTAGS`                    | Set RUNPATH, overriding RPATH, on linux platforms                     | `OFF`   |
-| `OPENASSETIO_ENABLE_COVERAGE`                     | Enable coverage reporting for gcc/clang                               | `OFF`   |
```

Comment:
Just a comment about unrelated changes; I keep having to go back to the commit list to figure out _why_ a change occurred that seems unrelated to improving C++ plugin support, e.g. why has OPENASSETIO_ENABLE_COVERAGE been removed?

Would it not be better to have this as part of a separate PR?

Reply:
Yeah I wrestled with this a bit. 

I've started deliberately adding ninja commits to fix things "whilst I'm in the neighbourhood", rather than create whole new issues/PRs, to reduce the process burden, given the reduced human resource working on the project at the moment.

I did assume most reviewers would review commit-by-commit, so it wouldn't be too annoying.

But if it's going to be a frustration for reviewers, then I can revisit this policy.

Reply:
I suspect with the size of the team, that we'll need to be more flexible.

A mention in the PR description of 'oh I've also fixed this, but it's independent' sort of comment would help that it's intentionally here amongst the real meat of the PR.

Reply:
I think I need to be as kind to reviewers as possible.  

I think a good middle ground is to continue (pragmatically) skipping creating issues, triage, estimation, scheduling, etc, and just do the work when I'm in the neighbourhood - but to at least split the work into a separate PR.  That's not _much_ more process burden (it does create possible merge conflicts and temporary CI failures), and is significantly easier to review when self contained and small.

With that in mind, I have removed the first two commits and put them up as separate PRs (#1361 and #1362).